### PR TITLE
Modular dimensions

### PR DESCRIPTION
**Pull request by dmcclean**
Sneak preview of #53, implemented in the context where #59 is accepted.

Still needs some module documentation before it should be merged, but this is a demonstration that the concept is sound, and that it imposes no API change on users of Numeric.Units.Dimensional.DK, whose export list remains completely unchanged.

Even the haddocks for Numeric.Units.Dimensional.DK remain unchanged, thanks to `{-# OPTIONS_HADDOCK not-home #-}` in the new modules, which hints to the haddock compiler that we would prefer to have links in other modules to these declarations go to Numeric.Units.Dimensional.DK and not to these new modules, which will only be directly imported for niche purposes by advanced users.

### CODE REVIEW

**Comment by dmcclean** [comment]
Additional bonus, this gives us a way to have arithmetic operators over term-level dimensions with their natural names, available for power users who might need them for computing with dimensions dynamically.

File: Numeric/Units/Dimensional/DK/Dimensions/TermLevel.hs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+import qualified Prelude as P
+
+-- | A physical dimension, encoded as 7 integers, representing a factorization of the dimension into the
+-- 7 SI base dimensions. By convention they are stored in the same order as 
+-- in the 'Numeric.Units.Dimensional.DK.Dimensions.TypeLevel.Dimension' data kind.
+data Dimension' = Dim' !Int !Int !Int !Int !Int !Int !Int 
+  deriving (Show, Eq, Ord)
+
+-- | The monoid of dimensions under multiplication.
+instance Monoid Dimension' where
```
Comment (bjornbm) [inline Numeric/Units/Dimensional/DK/Dimensions/TermLevel.hs]: I need to add `import Data.Monoid (Monoid(..))` for this to build. Another base-4.6 thing.

**Comment by dmcclean** [comment]
Sadly this causes a redundant import warning on 7.10 which is the only warning in the whole project. Is it worth `{-# LANGUAGE CPP #-}` or not?

**Comment by bjornbm** [comment]
Is that true if you change the Prelude import to:

    import Prelude (id, (+), (-), Int, Show, Eq, Ord)

?

**Comment by dmcclean** [comment]
Indeed not, nice call.

**Comment by bjornbm** [comment]
Ok. Looks sound to me. The term level Dimension' can indeed be useful. Ping me when this is ready for a merge and I can test on GHC 7.8 again.

**Comment by dmcclean** [comment]
OK, documentation is finished, this is fully baked as far as I can see.

**Pull request merged**